### PR TITLE
Integrate bravado-core for Swagger 2.0 response validation

### PR DESCRIPTION
**Pull request by analogue**

### CODE REVIEW

File: pyramid_swagger/tween.py [hunk 1]
```diff
@@ -131,7 +134,8 @@ def swagger_data(_):
             with validation_context(request, response=response):
                 swagger_handler.handle_response(
                     response,
-                    validator=getattr(op_or_validators_map, 'response', None))
+                    validator=getattr(op_or_validators_map, 'response', None),
+                    op=op_or_validators_map)
```
Comment (dnephin) [inline pyramid_swagger/tween.py]: To handle this better, could we change to interface of `handle_response()` to always just accept a single object `op_or_validators_map` ?

```python

swagger_handler.handle_response(response, op_or_validators_map)
```

That way any differences between them are hidden away behind the implementations

File: pyramid_swagger/tween.py [hunk 2]
```diff
@@ -456,9 +482,24 @@ def swaggerize_response(response, **kwargs):
...
+    try:
+        response_spec = get_response_spec(response.status_int, op)
+    except SwaggerMappingError:
+        error = ResponseValidationError(
+            "Could not find a matching Swagger response for {0} request {1}"
+            "with http_status {2}.".format(
+                op.http_method.upper(), op.path_name, response.status_code))
+
+        # Workaround for "W602 deprecated form of raising exception"
+        # See https://github.com/jcrocholl/pep8/issues/34
```
Comment (dnephin) [inline pyramid_swagger/tween.py]: Other options are:

* `#noqa` (https://pypi.python.org/pypi/flake8)
* https://pythonhosted.org/six/#six.reraise

File: requirements.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
 # TODO: Remove once bravado-core in pypi
--e git://github.com/analogue/bravado-core.git@0acc1415ed90a0ab52264b6267512891e193e3c5#egg=bravado-core
-# -e /nail/home/spatel/nfs/git/bravado-core-analogue
+-e git://github.com/Yelp/bravado-core.git@#egg=bravado-core
+#-e /nail/home/spatel/nfs/git/bravado-core-analogue
```
Comment (dnephin) [inline requirements.txt]: Delete

File: pyramid_swagger/tween.py (hunk 2 above)
Comment (analogue) [inline pyramid_swagger/tween.py]: `# noqa` didn't want to work for some reason. Had to use `flake8: noqa` to get it to actually ignore the warning.

File: requirements.txt (hunk 3 above)
Comment (analogue) [inline requirements.txt]: ack

File: pyramid_swagger/tween.py (hunk 1 above)
Comment (analogue) [inline pyramid_swagger/tween.py]: ack

**Comment by dnephin** [comment]
LGTM, ready for merge?

**Pull request merged**